### PR TITLE
feat!: Drop Node < 12, switch to eslint flat config

### PR DESCRIPTION
**Pull request by kecrily**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "author": "Vitor Balocco",
   "repository": "eslint/eslint-transforms",
   "engines": {
-    "node": ">=4"
+    "node": "^12.22.0 || ^14.17.0 || >=16.0.0"
```
Comment (mdjermanovic) [inline package.json:9]: This is a breaking change for this package.

File: package-lock.json
Comment (mdjermanovic) [inline package-lock.json:1]: This file shouldn't be checked-in. We can add it to `.gitignore`

File: package.json (hunk 1 above)
Comment (mdjermanovic) [reply package.json:9]: Which, I think, is fine but we should update the commit tag to `feat!` to bump major version on release.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Leaving open for others to review since this is a breaking change.

**Review by nzakas — APPROVED** [review]
LGTM. Because this tool is used rarely and in one-off situations, there's less concern over breaking changes.

**Pull request merged**